File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int>mp;
        for (int num: nums){
            mp[num]++;
        }
        for (int num: nums){
            if (mp[num] > 1) return 1;
        }
        return 0;
    }
};

File: Data Structures & Algorithms/permutations/submission-1.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        result = []
        temp = nums

        def solve(idx: int) -> None:
            if idx == len(nums):
                result.append(temp.copy())
                return
            
            for i in range(idx, len(nums)):

                temp[idx], temp[i] = temp[i], temp[idx]
                solve(idx + 1)
                temp[idx], temp[i] = temp[i], temp[idx]
        
        solve(0)
        return result
            

            